### ISSUE TITLE
The contaminant filter fails because the Docker image for BLAT (quay.io/biocontainers/blat:36--0) cannot be pulled 

### ISSUE THREAD

**Issue by egaffo** [nf-core/smrnaseq#354]
### Description of the bug

Enabling the contaminant filter the pipeline fails when pulling the blat docker container quay.io/biocontainers/blat:36--0.
The container image seems to be broken:
`docker run quay.io/biocontainers/blat:36--0` returns the error.

### Command used and terminal output

```console
nextflow run nf-core/smrnaseq \
        -profile docker \
        -resume \
        --input 'samplesheet.csv' \
        --genome 'GRCh38' \
        --mirtrace_species 'hsa' \
        --protocol 'illumina' \
        --outdir 'Results' \
        --fastp_max_length 35 \
        --filter_contamination true \
        --rrna 'human_rRNAs.fasta' \
        --trna 'human_tRNAs.fasta' \
        --pirna 'hsa_pirna.fa' \
        --other_contamination 'PhiX_and_adapter.fa' \
        --save_trimmed_fail true \
        --save_aligned true \
        --save_aligned_mirna_quant true \
        --save_merged true \
        --three_prime_adapter 'auto-detect'

ERROR ~ Error executing process > 'NFCORE_SMRNASEQ:CONTAMINANT_FILTER:BLAT_PIRNA (null)'

Caused by:
  Process `NFCORE_SMRNASEQ:CONTAMINANT_FILTER:BLAT_PIRNA (null)` terminated with an error exit status (125)

Command executed:

  echo other
  blat -out=blast8 hairpin.fa hsa_pirna.fa /dev/stdout | awk 'BEGIN{FS="        "}{if($11 < 1e-5)print $1;}' | uniq > mirnahit.txt
  awk 'BEGIN { while((getline<"mirnahit.txt")>0) l[">"$1]=1 } /^>/ {x = l[$1]} {if(!x) print }' hsa_pirna.fa  > filtered.fa
  
  cat <<-END_VERSIONS > versions.yml
  "NFCORE_SMRNASEQ:CONTAMINANT_FILTER:BLAT_PIRNA":
      blat: $(echo $(blat) | grep Standalone | awk '{ if (match($0,/[0-9]*[0-9]/,m)) print m[0] }')
  END_VERSIONS

Command exit status:
  125

Command output:
  (empty)

Command error:
  Unable to find image 'quay.io/biocontainers/blat:36--0' locally
  36--0: Pulling from biocontainers/blat
  docker: [DEPRECATION NOTICE] Docker Image Format v1 and Docker Image manifest version 2, schema 1 support is disabled by default and will be removed in an upcoming release. Suggest the author of quay.io/biocontainers/blat:36--0 to upgrade the image to the OCI Format or Docker Image manifest v2, schema 2. More information at https://docs.docker.com/go/deprecated-image-specs/.
  See 'docker run --help'.

Work dir:
  /sharedfs01/enrico/Joost/AGO2_circrnas/smrnaseq/work/d2/91d148e1932090db23675601dc03fe

Tip: when you have fixed the problem you can continue the execution adding the option `-resume` to the run command line

 -- Check '.nextflow.log' file for details
```

### System information

N E X T F L O W  ~  version 23.04.1 
Hardware  HPC
Executor SLURM
Container engine: Docker
OS: Ubuntu 22.04 LTS
nf-core/smrnaseq v2.3.1-g5901bea

**Comment by lpantano** [comment]
do we want to use this container instead:

```
community.wave.seqera.io/library/ucsc-blat:445--32730933d3c2c916
```

**Comment by apeltzer** [comment]
Maybe it would be worth to move the BLAT containers to nf-core/modules instead, importing them from there and making sure that they are already updated / matching the latest standards to improve quality.

**Comment by apeltzer** [comment]
As discussed: 

- First get the nf-tests up and running
- Then migrate to nf-core/modules
- Check container to use

**Comment by atrigila** [comment]
Closed via #397
